### DISCUSSION TITLE
Theme wrong

### DISCUSSION THREAD

**Discussion by sezanzeb** [wwmm/easyeffects#1475] [Q&A]
Hey, does anyone know why easyeffects would run with a different theme than the rest of the system? I think this is Adwaida or something, everything else uses Arc-Dark

Running gnome 41.4

**Comment by vchernin** [comment]
On Gnome 42 you can set dark theme in the settings app and that will make EasyEffects dark. 

Generally for libadwaita apps the gnome tweaks menu won’t work, but you can set `GTK_THEME`. However, that also needs the theme to support GTK4 + libadwaita. Note these aren’t technically supported by GTK since they can lead to broken looking apps, but it can still work.

FWIW for libadwaita there should be a recoloring API of some sort in the future.

**Reply by sezanzeb** [reply]
Thanks for your help.

The Arc theme supports GTK4:

```
➜  Arc-Dark pacman -Ss arc-gtk-theme
community/arc-gtk-theme 20220223-1 [installed]
    A flat theme with transparent elements for GTK 2,3,4 and Gnome-Shell
```

`GTK_THEME="Arc-Dark" easyeffects` still starts it with the adwaida theme.

**Reply by vchernin** [reply]
Strange that doesn't work, can you try copying the contents of `/usr/share/themes/Arc-Dark/gtk-4.0/gtk.css` to `~/.config/gtk-4.0/gtk.css`

**Reply by sezanzeb** [reply]
```
cp /usr/share/themes/Arc-Dark/gtk-4.0/gtk.css ~/.config/gtk-4.0
ls ~/.config/gtk-4.0 
gtk.css  settings.ini
```

no change

**Reply by vchernin** [reply]
is easyeffects being fully restarted? maybe reboot? at least one of those should have some impact.

**Reply by sezanzeb** [reply]
`GTK_THEME="Arc-Dark" easyeffects` works now, thanks. I haven't tried that one again since I opened this discussion, so I can't tell which step helped to get it to work.

Running easyeffects from the application menu still makes it use the Adwaida theme though.

**Reply by sezanzeb** [reply]
`sudo pkill -f -9 easyeffects` seems to be important.

After running easyeffects from the applications menu, closing it and then opening it with `GTK_THEME="Arc-Dark" easyeffects`, it will show up with the Adwaida theme again.

**Reply by vchernin** [reply]
> closing it and then opening it with GTK_THEME="Arc-Dark" easyeffects, it will show up with the Adwaida theme again.

That's because when you closed it, it only closed the window, it continues to run in the background. In the preferences page there is a setting "shutdown on window close" which ensures the app exits when closing the window.

**Reply by sezanzeb** [reply]
Interesting. How do you make the .desktop entry tell the existing process to open the window? Via dbus?

**Reply by vchernin** [reply]
>  How do you make the .desktop entry tell the existing process to open the window?

I am not familiar with the exact mechanism of how it "recognizes" the existing process and just opens a window.
But I do know this is indeed all done with D-bus services, which easyeffects uses since it passes `--gapplication-service`.

**Reply by vchernin** [reply]
If you want to set `GTK_THEME` for each launch though, you probably need to change the `exec` line in the .service file, not the .desktop file.

**Comment by wwmm** [comment]
>However, that also needs the theme to support GTK4 + libadwaita

That is problem one of the most important steps. Themes designed for gtk3 do not work with gtk4 applications
